s3storageprovider/src/main/java/org/duracloud/s3storage/S3StorageProvider.java

```diff
@@ -1020,4 +1033,39 @@ protected String getWithSpace(String name) {
         return name.replaceAll("%20", " ");
     }
 
+    static protected String encodeHeaderValue(String userMetaValue) {
+        try {
+            String encodedValue = HEADER_VALUE_PREFIX + URLEncoder.encode(userMetaValue, UTF_8);
+            return encodedValue;
+        } catch (UnsupportedEncodingException e) {
+            //this should never happen
+            throw new RuntimeException(e);
+        }
+    }
+
+    static protected String decodeHeaderValue(String userMetaValue) {
```

Comment:
There should be a test added which ensures that these encoding and decoding methods produce the expected results, especially when presented with non-ascii characters.